packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php

```diff
@@ -170,14 +170,17 @@ private function resolveAndSaveDependentFiles(
      * That's why we have to skip fatal errors of PHPStan caused by missing class,
      * so Rector can fix it first. Then run Rector again to refactor code with new classes.
      */
-    private function reportOrSkip(IdentifierNotFound $identifierNotFound): void
-    {
-        foreach ($this->renamedClassesDataCollector->getOldClasses() as $oldClass) {
-            if (str_contains($identifierNotFound->getMessage(), $oldClass)) {
-                return;
-            }
-        }
-
-        throw $identifierNotFound;
+    private function decoratePHPStanNodeScopeResolverWithRenamedClassSourceLocator(
+        NodeScopeResolver $nodeScopeResolver
+    ): void {
+        // 1. get PHPStan locator
+        /** @var ClassReflector $classReflector */
+        $classReflector = $this->privatesAccessor->getPrivateProperty($this->nodeScopeResolver, 'classReflector');
+        /** @var SourceLocator $sourceLocator */
+        $sourceLocator = $this->privatesAccessor->getPrivateProperty($classReflector, 'sourceLocator');
+
+        // 2. get Rector locator
+        $extendedSourceLocator = new AggregateSourceLocator([$sourceLocator, $this->renamedClassSourceLocator]);
+        $this->privatesAccessor->setPrivateProperty($classReflector, 'sourceLocator', $extendedSourceLocator);
     }
```

Comment:
@sabbelasichon Hi Sebastian, just to let you know, this is the fix we were looking for :) 

The `RenamedClassSourceLocator` needs more work, but we managed to get it into PHPStan.